src/Bicep.Core/SemanticModel/DeclaredSymbol.cs

```diff
@@ -25,11 +25,13 @@ protected DeclaredSymbol(ISymbolContext context, string name, SyntaxBase declari
         /// <summary>
         /// Gets the syntax node of the identifier.
         /// </summary>
-        public IdentifierSyntax NameSyntax { get; }
+        public IdentifierSyntaxBase NameSyntax { get; }
```

Comment:
I'm not sure I'm liking this. I think the alternative would be to allow IdentifierSyntax to have a list of tokens instead of just one.

Reply:
You could use SkippedTriviaSyntax to encapsulate a list of tokens btw (I think there are other places where we do this). Not sure if that's viable for your use case, just thought I'd throw it out there as a suggestion.

Reply:
Yeah the problem is that we use `IdentifierName` property of IdentifierSyntax all over the place, so `SkippedTriviaSyntax` will require adding extra checks all over the place. Having a class with the property is much more cleaner. However, I'm coming around to just modify `IdentifierSyntax` to capture a list of tokens instead of just one and have it return the right identifier name based on some conditions.

Reply:
Makes sense. I think we're saying the same thing; my suggestion was to modify `IdentifierSyntax` to have a child `SkippedTriviaSyntax | Token` rather than a list of tokens directly contained.

Reply:
Oh I see. I like that even better!

Reply:
It worked out pretty well actually.